modules/andes-core/broker/src/main/java/org/wso2/andes/server/cluster/HazelcastClusterAgent.java

```diff
@@ -195,6 +192,47 @@ public void start(ClusterManager manager) throws AndesException{
     }
 
     /**
+     * Check if the local node id is already taken by a different node in the cluster
+     *
+     * @param localMember
+     *         Current member
+     * @throws AndesException
+     */
+    private void checkForDuplicateNodeId(Member localMember) throws AndesException {
+        Set<Member> members = hazelcastInstance.getCluster().getMembers();
+
+        for (Member member : members) {
+            int nodeIdReadAttempts = 1;
+            String nodeIdOfMember = getIdOfNode(member);
+
+            /*
+             Node ID can be null if the node has not initialized yet. Therefore try to read the node id
+             MAX_NODE_ID_READ_ATTEMPTS times before failing.
+              */
+            while (nodeIdOfMember == null && nodeIdReadAttempts <= MAX_NODE_ID_READ_ATTEMPTS) {
```

Comment:
Shall we edit this like  ((nodeIdOfMember == null) && (nodeIdReadAttempts <= MAX_NODE_ID_READ_ATTEMPTS)) for readability purposes?